### PR TITLE
Trigger hda chain

### PR DESCRIPTION
**Pull request by makarukp**
- define more hda link inputs for mtl platform
- maintain flow by enabling pipeline trigger to start

### CODE REVIEW

**Review by abonislawski — APPROVED** [review]

**Review by ranj063 — APPROVED** [review]

File: src/ipc/ipc4/handler.c [hunk 1]
```diff
@@ -537,7 +537,8 @@ static int ipc4_process_chain_dma(struct ipc4_message_request *ipc4)
 		if (ret)
 			tr_err(&ipc_tr, "failed to create chain dma %d", ret);
 
-		return ret;
+		if (!cdma.primary.r.enable)
```
Comment (jsarha) [inline src/ipc/ipc4/handler.c:540]: Shouldn't we return also if there was on error? e.g.
if (!cdma.primary.r.enable || ret)

Comment (makarukp) [reply src/ipc/ipc4/handler.c]: Sure. I agree.

File: src/ipc/ipc4/handler.c [hunk 2]
```diff
@@ -537,7 +537,8 @@ static int ipc4_process_chain_dma(struct ipc4_message_request *ipc4)
 		if (ret)
 			tr_err(&ipc_tr, "failed to create chain dma %d", ret);
 
-		return ret;
+		if (!cdma.primary.r.enable || ret)
+			return ret;
```
Comment (kv2019i) [inline src/ipc/ipc4/handler.c:541]: For future PRs, please add a bit more text in git commit to explain the problem and why the fix helps. In what case the old code did not work, and how the patch addresses the problem. This makes the review process easier.

**Review by kv2019i — APPROVED** [review]

File: src/ipc/ipc4/handler.c (hunk 2 above)
Comment (lgirdwood) [reply src/ipc/ipc4/handler.c:541]: ack - can we have an inline comment here. Difficult to follow.

**Review by lgirdwood — COMMENTED** [review]
just need a comment to help understand the logic flow.

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**